xml/src/main/java/org/ehcache/xml/model/CacheSpec.java

```diff
@@ -133,53 +140,15 @@ public ListenersConfig listenersConfig() {
     return base != null ? new ListenersConfig(base, additionals.toArray(new ListenersType[0])) : null;
   }
 
-
   @Override
-  public Iterable<? extends ServiceConfiguration<?>> serviceConfigs() {
-    return sources.stream().flatMap(s -> s.getServiceConfiguration().stream()).map(this::parseCacheExtension)
-      .collect(Collectors.toMap(Object::getClass, identity(), (a, b) -> a)).values();
-  }
-
-  ServiceConfiguration<?> parseCacheExtension(final Element element) {
-    URI namespace = URI.create(element.getNamespaceURI());
-    final CacheServiceConfigurationParser<?> xmlConfigurationParser = serviceConfigParsers.get(namespace);
-    if(xmlConfigurationParser == null) {
-      throw new IllegalArgumentException("Can't find parser for namespace: " + namespace);
-    }
-    return xmlConfigurationParser.parseServiceConfiguration(element);
+  public Collection<Element> serviceConfigExtensions() {
+    return sources.stream().flatMap(s -> s.getServiceConfiguration().stream())
+      .collect(Collectors.toMap(Element::getTagName, identity(), (a, b) -> a)).values();
```

Comment:
Doesn't the cache definition override the template, and not the other way round?  Shouldn't it therefore be `(a, b) -> b`?